rewrite-yaml/src/test/java/org/openrewrite/yaml/CreateYamlFileTest.java

```diff
@@ -113,4 +132,76 @@ void shouldAddAnotherFile() {
           )
         );
     }
+
+    @Test
+    void shouldDownloadFileContents() {
+        @Language("yml")
+        String yamlContent = """
+          # This is a comment
+          foo: x
+          bar:
+            z: y
+          """;
+        InMemoryExecutionContext ctx = new InMemoryExecutionContext(e -> e.printStackTrace());
+        MockHttpSender httpSender = new MockHttpSender(() ->
+          new ByteArrayInputStream(yamlContent.getBytes()));
+        HttpSenderExecutionContextView.view(ctx)
+          .setHttpSender(httpSender)
+          .setLargeFileHttpSender(httpSender);
+        RemoteExecutionContextView.view(ctx).setArtifactCache(new RemoteArtifactCache() {
+            @Override
+            public @Nullable Path compute(URI uri, Callable<@Nullable InputStream> artifactStream, Consumer<Throwable> onError) {
+                return null;
+            }
+
+            @Override
+            public @Nullable Path get(URI uri) {
+                return null;
+            }
+
+            @Override
+            public @Nullable Path put(URI uri, InputStream is, Consumer<Throwable> onError) {
```

Comment:
Isn't the `put` the problem, because it returns null?
What about something like this:
```java
RemoteExecutionContextView.view(ctx).setArtifactCache(new RemoteArtifactCache() {
      
      @Override
      public @Nullable Path get(URI uri) {
          return null;
      }

      @Override
      public @Nullable Path put(URI uri, InputStream is, Consumer<Throwable> onError) {
          try {
              var file = File.createTempFile("rewrite", "yaml");
              Files.copy(is, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
              return file.toPath();
          } catch (IOException e) {
              throw new RuntimeException(e);
          }
      }
  });
```

Reply:
That indeed works, thanks a lot!